### PR TITLE
DOMA-3175 Message.uniqKey field with constraints for deduplication of notifications

### PR DESCRIPTION
**Pull request by akarjakin**
- added Message.uniqKey field
- added constraints
- tests for constraints
- some minor refactoring done.

### CODE REVIEW

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/notification/schema/Message.js [hunk 1]
```diff
@@ -106,6 +106,13 @@ const Message = new GQLListSchema('Message', {
             type: 'DateTimeUtc',
             isRequired: false,
         },
+
+        uniqKey: {
+            schemaDoc: 'Unique message key',
```
Comment (pahaz) [inline apps/condo/domains/notification/schema/Message.js:111]:
```suggestion
            schemaDoc: 'Unique message key. You can use it if you need to make sure that the message you are trying to create has not been created before. Fields `user`, `type` and `uniqkey` is to be unique. If you don\'t have a `user`, the fields `type` and `uniqkey` is to be unique',
```

File: apps/condo/domains/notification/schema/Message.test.js [hunk 2]
```diff
@@ -137,4 +147,78 @@ describe('Message', () => {
...
+                async () => await createTestMessage(admin, { user: { connect: { id: client.user.id } }, uniqKey }),
+                DUPLICATE_CONSTRAINT_VIOLATION_ERROR_MESSAGE,
+            )
+        })
+
+        it('checks Message constraints with non-null uniqKey and nullish user', async () => {
+            const admin = await makeLoggedInAdminClient()
+            const uniqKey = faker.datatype.uuid()
+
+            const [obj] = await createTestMessage(admin, { uniqKey })
```
Comment (pahaz) [inline apps/condo/domains/notification/schema/Message.test.js:172]: you need to pass `user: null` here! to protect your tests from changes inside the `createTestMessage`

File: apps/condo/domains/notification/schema/Message.test.js [hunk 3]
```diff
@@ -137,4 +147,78 @@ describe('Message', () => {
...
+
+            expect(obj.uniqKey).toMatch(uniqKey)
```
Comment (pahaz) [inline apps/condo/domains/notification/schema/Message.test.js:174]: you also need to check that the user toBeNull here

**Review by pahaz — APPROVED** [review]
You need also add the uniqKey to `SendMessageInput`

File: apps/condo/domains/notification/schema/Message.test.js (hunk 2 above)
Comment (akarjakin) [reply apps/condo/domains/notification/schema/Message.test.js:172]: done

File: apps/condo/domains/notification/schema/Message.test.js (hunk 3 above)
Comment (akarjakin) [reply apps/condo/domains/notification/schema/Message.test.js:174]: done

**Pull request merged**